crates/polars-time/src/upsample.rs

```diff
@@ -121,7 +121,9 @@ fn upsample_impl(
     stable: bool,
 ) -> PolarsResult<DataFrame> {
     let s = source.column(index_column)?;
-    s.ensure_sorted_arg("upsample")?;
+    if by.is_empty() {
+        s.ensure_sorted_arg("upsample")?;
+    }
```

Comment:
thanks for your PR!

does it work to move this check into `upsample_single_impl`?

Reply:
Thanks! That was indeed the cleaner way to do it.